aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js

```diff
@@ -625,26 +638,97 @@ define(["dojo/_base/declare",
                   username: null
                };
                var updatedConfig = lang.mixin(defaultConfig, config);
-
-               // Check to see whether or not the generated upload configuration indicates
-               // that an existing node will be created or not. If node is being updated then
-               // we need to generate an intermediary step to capture version and comments...
-               if (updatedConfig.overwrite === false)
+               
+               var walkFileSystem = lang.hitch(this, function alfresco_documentlibrary__AlfDndDocumentUploadMixin__onDndUploadDrop__walkFileSystem(directory, callback, error) {
```

Comment:
Normally we try to avoid nesting functions like this as this style means that it's not possible for them to be overridden by the widget that is mixing in the module. It also means that the functions themselves stay shorter which aids readability and also allows JSDoc'ing of the function.